httpx/_config.py

```diff
@@ -138,6 +134,14 @@ def _load_client_certs(self, cert: typing.Optional[CertTypes] = None) -> None:
                     password=cert[2],  # type: ignore
                 )
 
+    def __new__(
+        cls,
+        protocol: ssl._SSLMethod = ssl.PROTOCOL_TLS_CLIENT,
+        *args: typing.Any,
+        **kwargs: typing.Any,
+    ) -> "SSLContext":
+        return super().__new__(cls, protocol, *args, **kwargs)
```

Comment:
We need this to avoid "DeprecationWarning: ssl.SSLContext() without protocol argument is deprecated."